signature/jws/spec.go

```diff
@@ -4,46 +4,78 @@
 package jws
 
 import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
 	"time"
 
 	"github.com/golang-jwt/jwt/v4"
 	"github.com/notaryproject/notation-go"
+	"github.com/notaryproject/notation-go/spec/signature"
 )
 
-// unprotectedHeader contains the header parameters that are not integrity protected.
-type unprotectedHeader struct {
-	TimeStampToken []byte   `json:"timestamp,omitempty"`
-	CertChain      [][]byte `json:"x5c,omitempty"`
-}
-
-// MediaTypeNotationPayload describes the media type of the payload of notation signature.
-const MediaTypeNotationPayload = "application/vnd.cncf.notary.v2.jws.v1"
-
-// payload contains the subject manifest and other attributes that have to be integrity
-// protected.
-type payload struct {
-	Notation notationClaim `json:"notary"`
+type notaryClaim struct {
 	jwt.RegisteredClaims
-}
-
-// notationClaim is the top level node and private claim, encapsulating the notary v2 data.
-type notationClaim struct {
-	Subject notation.Descriptor `json:"subject"`
+	Notary signature.JWSNotaryClaim `json:"notary"`
 }
 
 // packPayload generates JWS payload according the signing content and options.
-func packPayload(desc notation.Descriptor, opts notation.SignOptions) *payload {
+func packPayload(desc signature.Descriptor, opts notation.SignOptions) jwt.Claims {
 	var expiresAt *jwt.NumericDate
 	if !opts.Expiry.IsZero() {
 		expiresAt = jwt.NewNumericDate(opts.Expiry)
 	}
-	return &payload{
-		Notation: notationClaim{
-			Subject: desc,
-		},
+	return notaryClaim{
 		RegisteredClaims: jwt.RegisteredClaims{
-			IssuedAt:  jwt.NewNumericDate(time.Now()),
 			ExpiresAt: expiresAt,
+			IssuedAt:  jwt.NewNumericDate(time.Now()),
 		},
+		Notary: signature.JWSNotaryClaim{Subject: desc},
```

Comment:
`Notary` claim should be renamed to `Subject`, there is typo in [signature spec](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#jws-json-serialization), addressed by this [PR](https://github.com/notaryproject/notaryproject/pull/156/files#diff-ab047165e5c22dc4d7380a85b336bd5ec76f2873057092648e583678a75a2c58R34)

Reply:
Done!